packages/flutter/lib/src/cupertino/sheet.dart

```diff
@@ -257,6 +257,10 @@ class CupertinoSheetTransition extends StatefulWidget {
     final bool isDarkMode = CupertinoTheme.brightnessOf(context) == Brightness.dark;
     final Color overlayColor = isDarkMode ? const Color(0xFFc8c8c8) : const Color(0xFF000000);
 
+    if (child != null && secondaryAnimation.isDismissed) {
+      return child;
+    }
```

Comment:
Could you explain why this fixes the bug for my own understanding?

Reply:
After coming back from sheet, if it shows `SlideTransition` which internally uses `RenderFractionalTranslation` which doesn't have size because its just a proxy box. So, throws the error as seen in #163954.

Also, it makes sense to just return page, when we have fully popped of from sheet.